lib/openssl/ecdsa.c

```diff
@@ -137,6 +137,19 @@ alg2crv(const char *alg)
     case 0: return "P-256";
     case 1: return "P-384";
     case 2: return "P-521";
+    case 3: return "secp256k1";
+    default: return NULL;
+    }
+}
+
+static const char *
+alg2hash(const char *alg)
+{
+    switch (str2enum(alg, NAMES, NULL)) {
+    case 0: return "S256";
+    case 1: return "S384";
+    case 2: return "S512";
+    case 3: return "S256";
```

Comment:
Is not there a missing "K" here? I mean, should not this be:
case 3: return "S256K";

Reply:
I guess I understand it now. You are not adding a new Hash function, so existing Hash algorithm (S256) will be reused.

Reply:
Exactly, and the former logic was relying on the alg name being just 1 char off the hash, so I had to add this mapping function.